odoo/addons/test_new_api/models/test_new_api.py

```diff
@@ -1147,3 +1147,23 @@ def _compute_container(self):
         container = self.env['test_new_api.compute.container']
         for member in self:
             member.container_id = container.search([('name', '=', member.name)], limit=1)
+
+
+class ConstrainedUnlinks(models.Model):
+    _name = 'test_new_api.model_constrained_unlinks'
+    _description = 'Model with unlink override that is constrained'
+
+    foo = fields.Char()
+    bar = fields.Integer()
+
+    @api.ondelete(at_uninstall=False)
```

Comment:
I think this should be the default behavior. This way, there wouldn't be a need to put `at_uninstall=False` almost every time you call @api.ondelete in each module. Only should be needed to put explicitly in case of `at_uninstall=True`.

Reply:
we made the conscious decision of making it as explicit as possible, so that the developer has to think twice whether their error should raise during a module uninstall (possibly crashing the registry / leaving crap in the DB) or not